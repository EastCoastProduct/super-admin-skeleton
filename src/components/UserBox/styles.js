import { StyleSheet } from 'aphrodite/no-important';
import { colors, sizes } from '../../styles/variables';

export default StyleSheet.create({
  header: {
    color: colors.header,
  },
  headerText: {
    textDecoration: 'underline',
  },
  box: {
    backgroundColor: colors.contentBcg,
    borderRadius: 3,
    boxShadow: `0 1px ${colors.shadow}`,
    marginTop: 20,
    minWidth: sizes.contentBox,
    padding: 40,
    width: '50%',
  },
});
