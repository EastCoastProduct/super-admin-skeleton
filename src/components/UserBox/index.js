import React, { PropTypes } from 'react';
import { css } from 'aphrodite/no-important';
import styles from './styles';

const UserBox = ({ children, header, profile }) =>
  <main>
    <h1 className={css(styles.header)}>
      <span className={css(styles.headerText)}>Users</span>
      {' > '}
      {(profile && (profile.get('firstname') || profile.get('lastname'))) ?
        `${profile.get('firstname')} ${profile.get('lastname')}` : header}
    </h1>
    <div className={css(styles.box)}>{children}</div>
  </main>;

UserBox.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.array,
  ]).isRequired,
  header: PropTypes.string.isRequired,
  profile: PropTypes.object,
};

export default UserBox;
