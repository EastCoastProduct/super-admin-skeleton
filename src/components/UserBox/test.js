import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import { fullProfile, profile } from '../../fixtures/user';
import deepMerge from 'deepmerge';
import UserBox, { getHeading } from './';

describe('UserBox component snapshot', () => {
  const props = {
    children: (
      <form key="1">
        <label>Form</label>
      </form>
    ),
    header: 'Edit Profile',
  };

  it('renders basic required data with single child', () => {
    const wrapper = shallow(<UserBox {...props} />);

    expect(wrapper).toMatchSnapshot();
    expect(getHeading(undefined, props.header)).toEqual(props.header);
  });

  it('renders empty profile data with array children', () => {
    const newP = deepMerge(props, {
      children: [
        props.children,
        <p key="2">Some text</p>
      ],
      profile: fromJS(profile),
    });
    const wrapper = shallow(<UserBox {...newP} />);

    expect(wrapper).toMatchSnapshot();
    expect(getHeading(newP.profile, newP.header)).toEqual(newP.header);
  });

  it('renders profile data with single child', () => {
    const newP = deepMerge(props, {
      profile: fromJS(fullProfile),
    });
    const wrapper = shallow(<UserBox {...newP} />);

    expect(wrapper).toMatchSnapshot();
    expect(getHeading(newP.profile, newP.header)).toEqual(
      `${newP.profile.get('firstname')} ${newP.profile.get('lastname')}`,
    );
  });

  it('renders profile data with just firstname and single child', () => {
    const newP = deepMerge(props, {
      profile: fromJS(profile).set('firstname', 'John'),
    });
    const wrapper = shallow(<UserBox {...newP} />);

    expect(wrapper).toMatchSnapshot();
    expect(getHeading(newP.profile, newP.header))
      .toEqual(newP.profile.get('firstname'));
  });

  it('renders profile data with just lastname and single child', () => {
    const newP = deepMerge(props, {
      profile: fromJS(profile).set('lastname', 'Doe'),
    });
    const wrapper = shallow(<UserBox {...newP} />);

    expect(wrapper).toMatchSnapshot();
    expect(getHeading(newP.profile, newP.header))
      .toEqual(newP.profile.get('lastname'));
  });
});
