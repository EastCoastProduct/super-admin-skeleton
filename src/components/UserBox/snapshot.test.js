import React from 'react';
import renderer from 'react-test-renderer';
import { fromJS } from 'immutable';
import { StyleSheetTestUtils } from 'aphrodite/no-important';
import UserBox from './';

describe('UserBox component snapshot', () => {
  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });
  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('renders basic required data with single child', () => {
    const tree = renderer.create(
      <UserBox header="View Profile">
        <form>
          <label>Form</label>
        </form>
      </UserBox>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders basic required data with array children', () => {
    const tree = renderer.create(
      <UserBox header="View Profile">
        <form>
          <label>Form</label>
        </form>
        <p>Some text</p>
      </UserBox>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders profile data with sinlge child', () => {
    const profile = fromJS({
      firstname: 'John',
      lastname: 'Doe',
    });
    const tree = renderer.create(
      <UserBox header="View Profile" profile={profile}>
        <form>
          <label>Form</label>
        </form>
      </UserBox>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
