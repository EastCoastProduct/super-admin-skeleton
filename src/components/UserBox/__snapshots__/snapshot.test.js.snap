exports[`UserBox component snapshot renders basic required data with array children 1`] = `
<main>
  <h1
    className="header_fffb07">
    <span
      className="headerText_104b05c">
      Users
    </span>
     > 
    View Profile
  </h1>
  <div
    className="box_5m47zw">
    <form>
      <label>
        Form
      </label>
    </form>
    <p>
      Some text
    </p>
  </div>
</main>
`;

exports[`UserBox component snapshot renders basic required data with single child 1`] = `
<main>
  <h1
    className="header_fffb07">
    <span
      className="headerText_104b05c">
      Users
    </span>
     > 
    View Profile
  </h1>
  <div
    className="box_5m47zw">
    <form>
      <label>
        Form
      </label>
    </form>
  </div>
</main>
`;

exports[`UserBox component snapshot renders profile data with sinlge child 1`] = `
<main>
  <h1
    className="header_fffb07">
    <span
      className="headerText_104b05c">
      Users
    </span>
     > 
    John Doe
  </h1>
  <div
    className="box_5m47zw">
    <form>
      <label>
        Form
      </label>
    </form>
  </div>
</main>
`;
